expo/expo
# [av] Fix precision of the player position

# Why

Hello, Expo Team!

We are developing a complex video editor for our client, utilizing your `EXAVPlayer`. This editor enables users to seek any millisecond of the video. However, we encountered some issues while attempting to use `setPositionAsync`.

# How

As per Apple's documentation (https://developer.apple.com/documentation/avfoundation/avplayer/1387741-seektotime?language=objc), we understand that `toleranceBefore` and `toleranceAfter` can be set. Despite this, we've noticed edge cases where EXAVPlayer fails to navigate to the specified millisecond.

# Test Plan

The issue was tested in a local project. Adjusting the precision resolved the problem related to rounding.

Before:
1. Load any video into the `<Video />` tag, providing a reference (`videoRef`).
2. Execute the following:

```tsx
videoRef.current?.setPositionAsync(267, {
    toleranceMillisBefore: 0,
    toleranceMillisAfter: 0
})
```
3.  Observe `newPosition` in Objective-C.

![image](https://github.com/expo/expo/assets/9088288/37ce99d9-c564-467f-86ed-58cc8ef504bf)

4. Notice the video did not seek to 267 ms

After:

1. Load any video into the `<Video />` tag, providing a reference (`videoRef`).
2. Execute the following:

```tsx
videoRef.current?.setPositionAsync(267, {
    toleranceMillisBefore: 0,
    toleranceMillisAfter: 0
})
```
3.  Observe `newPosition` in Objective-C.

![image](https://github.com/expo/expo/assets/9088288/eb774e05-3fe1-4fa7-9702-ba36a9e74c1a)


4. The video now seeks accurately to 267 ms.

# Checklist

No need to update the docs
